javaschool-backend/src/main/java/com/nearsoft/javaschoolbackend/service/impl/PackageInfoServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.nearsoft.javaschoolbackend.service.impl;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nearsoft.javaschoolbackend.model.response.PackageSize;
+import com.nearsoft.javaschoolbackend.model.response.PackageType;
+import com.nearsoft.javaschoolbackend.service.PackageInfoService;
+import com.nearsoft.javaschoolbackend.util.RabbitMQSender;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class PackageInfoServiceImpl implements PackageInfoService {
+
+    private final RabbitMQSender rabbitMQSender;
+
+    private final ObjectMapper objectMapper;
+
+    public PackageInfoServiceImpl(RabbitMQSender rabbitMQSender, ObjectMapper objectMapper){
+        this.rabbitMQSender = rabbitMQSender;
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public List<String> getPackageTypes() {
+        List<PackageType> packageTypes = new ArrayList<PackageType>();
+        try {
+            packageTypes = objectMapper.readValue(rabbitMQSender.send("packageType"), new TypeReference<List<PackageType>>(){});
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        List<String> res = new ArrayList<String>();
+        for(PackageType type: packageTypes){
+            res.add(type.getDescription());
```

Comment:
it's better to minimize changing the inner state of an object/scope, so instead of creating a mutable list like sizesName you can do res.stream().map ... to achieve the same but with immutable processes.